### PR TITLE
Initial Account Setup

### LINKED ISSUE #9: Account Options
Rather than manually handling passwords: 

- could use google OAuth as authentication service
  - in theory could probably offer multiple oath options
- could use phone number & 2fa as entire login (would need a service to send sms though - probably not free)
- or open to other options

### PR DESCRIPTION
**Pull request by jaredwebber**
Added login screen, initial google oauth setup (almost entirely from [here](https://medium.com/@marwa.diab/firebase-authentication-in-swiftui-part-1-71a409108d9f))

Remaining:
- Actually link firebase authenticated accounts with custom User objects
- Apple authentication
- Displaying user-relevant notes instead of all notes ever

Closes #9

### CODE REVIEW

File: BeSpoke/Views/SharedViews/LoginView.swift [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                    onCompletion: { _ in
+                        // TODO: Handle AppleID Completion
+                    }
+                )
+                .signInWithAppleButtonStyle(colorScheme == .light ? .black : .white)
+                .frame(width: 280, height: 45, alignment: .center)
+
+                GoogleSignInButton {
+                    signInWithGoogle()
+                    print("Sign In With Google")
```
Comment (wrighted) [inline BeSpoke/Views/SharedViews/LoginView.swift:43]: Nit: Is this just for logging purposes? Maybe "Signed in with Google" then?

**Review by wrighted — APPROVED** [review]
Looks good to me, signed in and added a note.

Since the apps idea is up in the air I'm okay with having the "plus circle" for now, but someone should figure out what our gimmick is.

Other than that seems to work good job keep it up.

File: BeSpoke/Views/SharedViews/LoginView.swift (hunk 1 above)
Comment (jaredwebber) [reply BeSpoke/Views/SharedViews/LoginView.swift:43]: Thats just leftover from before the button did anything we can get rid of it I think

**Comment by jaredwebber** [comment]
> Since the apps idea is up in the air I'm okay with having the "plus circle" for now, but someone should figure out what our gimmick is.

Ya not sure what the plan is, just easier to test when adding a note is just a button instead of having to mess with current date / first login

**Pull request merged**